core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -534,66 +536,50 @@ private void updateAdditionalResponseHeaders(
         }
     }
 
-    @Override
-    public boolean removeAdditionalResponseHeader(CharSequence name) {
-        return removeAdditionalResponseHeader(additionalResponseHeadersUpdater, name);
-    }
-
-    private boolean removeAdditionalResponseHeader(
-            AtomicReferenceFieldUpdater<DefaultServiceRequestContext, HttpHeaders> atomicUpdater,
-            CharSequence name) {
-        requireNonNull(name, "name");
-        for (;;) {
-            final HttpHeaders oldValue = atomicUpdater.get(this);
-            if (oldValue.isEmpty() || !oldValue.contains(name)) {
-                return false;
-            }
-
-            final HttpHeaders newValue = oldValue.toBuilder().removeAndThen(name).build();
-            if (atomicUpdater.compareAndSet(this, oldValue, newValue)) {
-                return true;
-            }
-        }
-    }
-
     @Override
     public HttpHeaders additionalResponseTrailers() {
         return additionalResponseTrailers;
     }
 
+    /**
+     * Returns the {@link HttpHeaders} which is set using {@code (add|set)AdditionalResponseTrailer} and
+     * clears the {@link HttpHeaders} so that the {@link HttpHeaders} is not sending twice when
+     * a {@link Service} completes an {@link HttpResponse}.
+     *
+     * <p>This is used only when the trailers is sent as a {@link ResponseHeaders} in gRPC.
+     */
+    public HttpHeaders getAndRemoveAdditionalResponseTrailers() {
+        return additionalResponseTrailersUpdater.getAndSet(this, HttpHeaders.of());
+    }
```

Comment:
We will not need this method at all if we have `mutateAdditionalResponseTrailers()`:

```java
HttpHeaders oldTrailers = ctx.additionalResponseTrailers();
ctx.mutateAdditionalResponseTrailers(mutator -> mutator.clear());
```